apps/studio/src/features/editing-experience/components/MoveResourceModal/MoveItem.tsx

```diff
@@ -0,0 +1,85 @@
+import { Suspense, useMemo } from "react"
+import { Text } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+import { QueryErrorResetBoundary } from "@tanstack/react-query"
+import { useAtomValue } from "jotai"
+import { ErrorBoundary } from "react-error-boundary"
+import { BiFile, BiFolder, BiHome } from "react-icons/bi"
+
+import type { RouterOutput } from "~/utils/trpc"
+import { moveResourceAtom } from "../../atoms"
+
+type MoveItemProps = Pick<
+  RouterOutput["resource"]["getChildrenOf"][number],
+  "permalink" | "type"
+> & {
+  onChangeResourceId: () => void
+  resourceId: string
+}
+
+const SuspendableMoveItem = ({
+  permalink,
+  resourceId,
+  type,
+  onChangeResourceId,
+}: MoveItemProps) => {
+  const movedItem = useAtomValue(moveResourceAtom)
+  const icon: JSX.Element = useMemo(() => {
+    switch (type) {
+      case "Folder":
+        return <BiFolder />
+      case "Page": {
+        return permalink === "/" ? <BiHome /> : <BiFile />
+      }
+    }
+  }, [permalink, type])
+
+  return (
+    <Button
+      variant="clear"
+      w="full"
+      justifyContent="flex-start"
+      color="base.content.default"
+      // NOTE: Homepage is special - it's a page but
+      // we also use it to represent site root
+      isDisabled={
+        (type === "Page" && permalink !== "/") ||
+        resourceId === movedItem?.resourceId
+      }
+      // NOTE: Set special styling because pages should show normally
+      // but have no onClick
+      // TODO: add permissions for folders that users have no access to
+      // These should have disabled styling
+      _disabled={{ color: "base.content.default" }}
+      pl="2.25rem"
+      onClick={onChangeResourceId}
+    >
+      {icon}
```

Comment:
nit: buttons have a `leftIcon` prop that can be used for the correct styling:

```suggestion
    <Button
      variant="clear"
      w="full"
      justifyContent="flex-start"
      color="base.content.default"
      // NOTE: Homepage is special - it's a page but
      // we also use it to represent site root
      isDisabled={
        (type === "Page" && permalink !== "/") ||
        resourceId === movedItem?.resourceId
      }
      // NOTE: Set special styling because pages should show normally
      // but have no onClick
      // TODO: add permissions for folders that users have no access to
      // These should have disabled styling
      _disabled={{ color: "base.content.default" }}
      pl="2.25rem"
      onClick={onChangeResourceId}
      leftIcon={icon}
    >
```

Reply:
done